src/firefly/js/visualize/ui/ExtractionDialog.jsx

```diff
@@ -448,6 +451,37 @@ function ExtractionPanelView({pointSize, setPointSize, afterRedraw, plotlyDivSty
                                  plotlyData, canCreateExtractionTable,
                                  plotlyLayout, startUpHelp, callKeepExtraction,
                                  bottomUI, combineOp, setCombineOp, hasFloatingData}) {
+
+    const extractAddedChartInfo = (action) => {
+        const chartId = action.payload.chartId;
+        const tbl_id =   action.payload.data?.[0].tbl_id;
+        return {chartId, tbl_id};
+    };
+
+    const doPinChart = () => {
+        const {tbl_id} = callKeepExtraction(false, true);
+        // when a new table is added, an 'active chart' will be added when ChartContainer is mounted.
+        // watch for chart_add to extract chart_id, then pin it.
+        dispatchAddActionWatcher({actions: [TABLE_UPDATE, CHART_ADD], callback: (action, cancelSelf) => {
+                const aType = action?.type;
+                if (aType === CHART_ADD) {
+                    const {chartId, tbl_id:cTblId} = extractAddedChartInfo(action);
+                    if (cTblId === tbl_id) {  // only pin the chart from the table that just got created
+                        cancelSelf?.();
+                        pinChart({chartId});
+                    }
+                    cancelSelf?.();
+                } else {
+                    // this flow assumes CHART_ADD will happen immediately after table is pinned
+                    // have a fallback to cancel this watcher in cases when this is not true.
+                    // since we have tbl_id, we will watch for any TABLE_UPDATE to catch cases where CHART_ADD was not called
+                    const aTblId = action?.payload?.tbl_id;
+                    if (aTblId !== tbl_id) cancelSelf?.();
+                }
+            }
+        });
+    };
+
```

Comment:
I don't think this function should be a closure. I makes the component hard to read. 
just make a function to take `callKeepExtraction` as a parameter.
```js
function doPinChart(callKeepExtraction) {
```